chain/sync.go

```diff
@@ -1301,31 +1301,33 @@ loop:
 		at = blks[len(blks)-1].Parents()
 	}
 
-	// base is the tipset in the candidate chain at the height equal to our known tipset height.
-	if base := blockSet[len(blockSet)-1]; !types.CidArrsEqual(base.Parents().Cids(), known.Cids()) {
-		if base.Parents() == known.Parents() {
-			// common case: receiving a block thats potentially part of the same tipset as our best block
-			return blockSet, nil
-		}
+	base := blockSet[len(blockSet)-1]
+	if base.Parents() == known.Parents() {
+		// common case: receiving a block thats potentially part of the same tipset as our best block
+		return blockSet, nil
+	}
 
-		// We have now ascertained that this is *not* a 'fast forward'
+	if types.CidArrsEqual(base.Parents().Cids(), known.Cids()) {
```

Comment:
This and the previous check seem to be the same now, I'd keep this one that uses the API and avoid comparing structs directly.

Reply:
You mean keep `CidArrsEqual`?
The checks are not the same, one has one less `.Parent()`.

Reply:
You're right, missed that.